src/controllers/searchController.js

```diff
@@ -0,0 +1,132 @@
+        },
+        include: [
+          {
+            model: Article,
+            as: 'articles',
+            attributes: ['slug', 'title', 'description', 'body', 'imgUrl']
+          }
+        ],
+        attributes: ['id', 'username', 'image', 'email', 'image']
+      });
+      if (author) {
+        if (author.dataValues.articles.length < 1) {
+          return response.status(404).json(errorResponse(['No article found']));
+        }
+        return response.status(200).json({
+          status: 'Success',
+          message: 'Articles found successfully',
+          author
+        });
+      }
+      return response.status(404).json(errorResponse(['Author not found']));
+    } catch (error) {
+      return response.status(500).json(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description - This method is responsible for creating new articles
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   * @returns {object} - object representing response message
+   * @memberof SearchController
+   */
+  static async searchByKeyword(request, response) {
+    const { keyword } = request.params;
+    try {
+      const articles = await Article.findAll({
+        where: {
+          $or: [
+            {
+              title: { $iLike: `%${keyword}%` }
+            },
+            {
+              description: { $iLike: `%${keyword}%` }
+            }
+          ]
+        },
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: ['id', 'username', 'email', 'image']
+          }
+        ],
+        attributes: ['slug', 'title', 'description', 'body', 'imgUrl', 'readtime']
+      });
+      if (articles.length < 1) {
+        return response.status(404).json(errorResponse(['No article found']));
+      }
+      return response.status(200).json({
+        status: 'Success',
+        message: 'Articles found successfully',
+        articles
+      });
+    } catch (error) {
+      return response.status(500).json(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description - This method is responsible for searching for articles by tags
+   * @param {object} request - Request object
+   * @param {object} response - Response object
+   * @returns {object} - object representing response messages
+   * @memberof SearchController
+   */
+  static async searchByTags(request, response) {
+    const tagName = request.params.tagname;
+    const tagModified = tagName.toLowerCase();
+    try {
+      const articles = await Tag.findOne({
+        where: {
+          name: tagModified
+        },
+        include: [
+          {
+            model: Article,
+            as: 'articles',
+            attributes: ['slug', 'title', 'description', 'body', 'imgUrl', 'readtime']
+          }
+        ]
+      });
+      if (articles === null || articles.length === 0) {
```

Comment:
Please what difference would it make if you had used `articles.length === 0` alone versus using `articles === null || articles.length === 0` ?

Reply:
Hi @marcdomain, nice observation. On second thought I think there is a need to refactor that aspect. There isn't much of a difference.

Reply:
@marcdomain, I have refactored it.